src/BoundingBoxCommand.cc

```diff
@@ -0,0 +1,382 @@
+
+    Json::Value props = get_value<Json::Value>(cmd, "properties");
+    // props[VDMS_ROI_SHAPE_PROP] = ( rect > -1 ) ? "rectangle" : "polygon";
+
+    if (rect > -1) {
+        props["_x1"] = get_value<int>(rect, "x");
+        props["_y1"] = get_value<int>(rect, "y");
+        props["_width"] = get_value<int>(rect, "width");
+        props["_height"] = get_value<int>(rect, "height");
+    }
+    // else if (poly > -1) {
+    //     for (int i = 0; i < poly.size(); ++i ) {
+    //         std::string x = "_x" + i;
+    //         std::string y = "_y" + i;
+    //         props[x] = get_value<int>(poly[i], "x");
+    //         props[y] = get_value<int>(poly[i], "y");
+    //     }
+    // }
+    else {
+        error["info"] = "Coordinates not properly specified";
+        error["status"] = RSCommand::Error;
+        return -1;
+    }
+
+    // Add Region node
+    query.AddNode(node_ref, VDMS_ROI_TAG, props, Json::Value());
+
+    if ( cmd.isMember("image") ) {
+        Json::Value img;
+        img["ref"] = cmd["image"];
+        add_link(query, img, node_ref, VDMS_ROI_EDGE_TAG);
+    }
+
+    return 0;
+}
+
+//========= UpdateBoundingBox definitions =========
+
+UpdateBoundingBox::UpdateBoundingBox() : BoundingBoxCommand("UpdateBoundingBox")
+{
+}
+
+int UpdateBoundingBox::construct_protobuf(PMGDQuery& query,
+    const Json::Value& jsoncmd,
+    const std::string& blob,
+    int grp_id,
+    Json::Value& error)
+{
+    const Json::Value& cmd = jsoncmd[_cmd_name];
+
+    // Update Image node
+    query.UpdateNode(get_value<int>(cmd, "_ref", -1),
+                     VDMS_ROI_TAG,
+                     cmd["properties"],
+                     cmd["remove_props"],
+                     cmd["constraints"],
+                     get_value<bool>(cmd, "unique", false));
+
+    return 0;
+}
+
+//========= FindBoundingBox definitions =========
+
+FindBoundingBox::FindBoundingBox() : BoundingBoxCommand("FindBoundingBox")
+{
+}
+
+int FindBoundingBox::construct_protobuf(
+    PMGDQuery& query,
+    const Json::Value& jsoncmd,
+    const std::string& blob,
+    int grp_id,
+    Json::Value& error)
+{
+    const Json::Value& cmd = jsoncmd[_cmd_name];
+
+    // if blob is true, make sure we have a reference, that way we can iterate
+    // over the bounding boxes and find the link to the image (if it exists)
+    Json::Value results = get_value<Json::Value>(cmd, "results");
+    int ref = get_value<int>(cmd, "_ref", query.get_available_reference());
+
+    bool coords = false;
+    if (results.isMember("list")) {
+        for (int i = 0; i < results["list"].size(); ++i) {
+            if (results["list"][i].asString() == "_coordinates") {
+                Json::Value aux;
+                results["list"].removeIndex(i, &aux);
+                coords = true;
+                break;
+            }
+        }
+    }
+
+    if (get_value<bool>(results, "blob"))
+        coords = true;
+
+    if (coords) {
```

Comment:
The variable coords can be set either if the user asks for _coordinates as part of the result block, or if the user requests a blob. So there are two places it can be set. I'm fine with doing something else if you'd like, but this seemed like the most straightforward way to handle it.